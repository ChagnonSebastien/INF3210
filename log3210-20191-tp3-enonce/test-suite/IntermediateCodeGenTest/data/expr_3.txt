Declaration
bool a;
bool b;
bool c;
num d;
num e;
num f;
EndDeclaration

//expression with variable

a = true || false;

b = true || a;

c = a && b;


d = 1 + 2 + 3;

e = d * 1;

f = e - -d;