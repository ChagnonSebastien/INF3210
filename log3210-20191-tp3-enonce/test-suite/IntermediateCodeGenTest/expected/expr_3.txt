goto _L0
goto _L1
_L0
a = 1
goto _L2
_L1
a = 0
_L2
goto _L3
if a == 1 goto _L3
goto _L4
_L3
b = 1
goto _L5
_L4
b = 0
_L5
ifFalse a == 1 goto _L7
if b == 1 goto _L6
goto _L7
_L6
c = 1
goto _L8
_L7
c = 0
_L8
_t0 = 1 + 2
_t1 = _t0 + 3
d = _t1
_t2 = d * 1
e = _t2
_t3 = - d
_t4 = e - _t3
f = _t4
